### PR TITLE
Update PyTorch pin to the version which support `elapsed_time`; remove our patch for `elapsed_time`

### PR DESCRIPTION
**Pull request by anmyachev**
PyTorch commit:
* https://github.com/pytorch/pytorch/commit/77748ed8ec2dea0deb6ffd59bd28329e6cdc9d56
* https://github.com/pytorch/pytorch/commit/8dd4673ceacad362f7c5c4bacdd52d64e2a9f601
* https://github.com/pytorch/pytorch/commit/61dc5e9c0a36d590adc47b4110efd94d9eb59306 - https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12242668384 (passed)
* https://github.com/pytorch/pytorch/commit/2d708752f034118b14bf5b7b3644eb684c5cc8cc - https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12234429829 (`RuntimeError: Overflow when unpacking long`)

Inductor CI:
* https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12196498158
* https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12200420024 (another commit, failed because of https://github.com/pytorch/pytorch/issues/142102)
* https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12202116045 (issue with icpx)
* https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12203639957 (passed with patch)

> The impact on the Triton tutorials performance measurements should be assessed prior to the patch deprecation. In case if any regressions and/or new issues the mitigations should be implemented prior to the patch deprecation.

According to https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12242518858/job/34150030016 it's fine to go.

### CODE REVIEW

**Comment by anmyachev** [comment]
@chengjunlu there are many messages like: `warning: Double arithmetic operation is not supported on this platform with FP64 conversion emulation mode (poison FP64 kernels is enabled).` in https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12196498158/job/34024201713. They really clutter up the logs and don't provide much information. It would be great to remove them somehow. Is this possible?

**Comment by pbchekin** [comment]
> @pbchekin do you know why? (from https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12196498158)
> 
> ```shell
> FAILED [4.7235s] inductor/test_triton_kernels.py::CustomOpTests::test_autotune_unbacked - torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
> FileNotFoundError: [Errno 2] No such file or directory: '/opt/intel/oneapi/bin/icpx'
> ```

There is no such file, indeed. This file is located at:

* `/opt/intel/oneapi/compiler/2024.1/bin/icpx`, `/opt/intel/oneapi/pytorch-gpu-dev-0.5/bin/icpx` in PTDB
* `/opt/intel/oneapi/2025.0/bin/icpx`, `/opt/intel/oneapi/compiler/2025.0/bin/icpx` in DLE

**Comment by anmyachev** [comment]
> > @pbchekin do you know why? (from https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12196498158)
> > ```shell
> > FAILED [4.7235s] inductor/test_triton_kernels.py::CustomOpTests::test_autotune_unbacked - torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
> > FileNotFoundError: [Errno 2] No such file or directory: '/opt/intel/oneapi/bin/icpx'
> > ```
> 
> There is no such file, indeed. This file is located at:
> 
> * `/opt/intel/oneapi/compiler/2024.1/bin/icpx`, `/opt/intel/oneapi/pytorch-gpu-dev-0.5/bin/icpx` in PTDB
> * `/opt/intel/oneapi/2025.0/bin/icpx`, `/opt/intel/oneapi/compiler/2025.0/bin/icpx` in DLE

I found the reason - it's because of how pytorch searches for sycl home and it probably relates to https://github.com/pytorch/pytorch/commit/4742080ed986b03b21b6ef2b16c03e77e4b391b8.

Ref to PyTorch: https://github.com/pytorch/pytorch/blame/5872a8c6b00a5c9e45ac4bc99a5c87b93a93aa94/torch/utils/cpp_extension.py#L147
```python
def _find_sycl_home() -> Optional[str]:
    """Find the OneAPI install path."""
    # Guess #1
    sycl_home = os.environ.get('ONEAPI_ROOT')
    if sycl_home is None:
        # Guess #2
        icpx_path = shutil.which('icpx')
        if icpx_path is not None:
            sycl_home = os.path.dirname(os.path.dirname(
                os.path.realpath(icpx_path)))

    if sycl_home and not torch.xpu.is_available():
        print(f"No XPU runtime is found, using ONEAPI_ROOT='{sycl_home}'",
              file=sys.stderr)
    return sycl_home
```

**Comment by anmyachev** [comment]
@pbchekin any chance that `ONEAPI_ROOT` for PTDB is `/opt/intel/oneapi/2025.0` instead of `/opt/intel/oneapi` (for DLE)? That would explain the error.

My guess comes from https://github.com/pytorch/pytorch/pull/142242#discussion_r1874209890.

**Comment by pbchekin** [comment]
> @pbchekin any chance that `ONEAPI_ROOT` for PTDB is `/opt/intel/oneapi/2025.0` instead of `/opt/intel/oneapi` (for DLE)? That would explain the error.
> 
> My guess comes from [pytorch/pytorch#142242 (comment)](https://github.com/pytorch/pytorch/pull/142242#discussion_r1874209890).

Nope:

```
$ source /opt/intel/oneapi/setvars.sh
...
$ echo $ONEAPI_ROOT
/opt/intel/oneapi
```

**Comment by anmyachev** [comment]
Inductor CI with changes from PR 2962: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12221768210

**Comment by chengjunlu** [comment]
> @chengjunlu there are many messages like: `warning: Double arithmetic operation is not supported on this platform with FP64 conversion emulation mode (poison FP64 kernels is enabled).` in https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12196498158/job/34024201713. They really clutter up the logs and don't provide much information. It would be great to remove them somehow. Is this possible?

This warning information is a DPC++ feature. Let me check with torch team how to disable it.

File: .github/pins/pytorch-upstream.txt [hunk 1]
```diff
@@ -1 +1 @@
-d0fd42eb3ac6939e63879bc055b6901103f713d3
+61dc5e9c0a36d590adc47b4110efd94d9eb59306
```
Comment (anmyachev) [inline .github/pins/pytorch-upstream.txt:1]:
```suggestion
2d708752f034118b14bf5b7b3644eb684c5cc8cc
```

**Comment by anmyachev** [comment]
Hi @guangyey!

After this change https://github.com/pytorch/pytorch/pull/135567 our tutorials started to fail with `RuntimeError: Overflow when unpacking long` exception (ref: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12239696322/job/34140774158?pr=2952).

A quick search through the PyTorch codebase gave me the idea that the problem in `infer_scalar_type` function https://github.com/pytorch/pytorch/blob/90fc2b42e3e2d51b26a96df0dff4a644e218f8ab/torch/csrc/utils/tensor_new.cpp#L148. Returning type `ScalarType:Long` instead of type `ScalarType::Uint64`. Could you take a look?

Example:
```python
# Each pointer is obtained through `tensor.data_ptr()` function.
d_a_ptrs = torch.tensor([18374686479673720832, 18374967954644140032, 18374967954645188608, 18374967954645450752], device=device) # <- failed
```

**Comment by guangyey** [comment]
> I decided to try specifying the dtype directly, as suggested in [pytorch/pytorch#135628](https://github.com/pytorch/pytorch/issues/135628). @guangyey do I understand correctly that it is now recommended to use this approach in the code?

In my understanding, specifying the dtype is the correct approach. This is because we change `data_ptr` from int64 to uint64 in https://github.com/pytorch/pytorch/pull/135567

**Review by guangyey — APPROVED** [review]
LGTM.

**Pull request merged**